api/src/main/java/org/openmrs/util/OpenmrsUtil.java

```diff
@@ -812,16 +783,22 @@ public static int compareWithNullAsLatest(Date d1, Date d2) {
 		return ret;
 	}
 	
-	// TODO: properly handle duplicates
 	public static List<Concept> conceptListHelper(String descriptor) {
 		List<Concept> ret = new ArrayList<>();
+		Set<String> duplicates = new HashSet<>();
 		if (descriptor == null || descriptor.length() == 0) {
 			return ret;
 		}
 		ConceptService cs = Context.getConceptService();
 		
 		for (StringTokenizer st = new StringTokenizer(descriptor, "|"); st.hasMoreTokens();) {
 			String s = st.nextToken().trim();
+			if(duplicates.contains(s)) {
+				continue;
+			}
+			else {
+			duplicates.add(s);
```

Comment:
* Can you correct this indention?
* By the way, why do we need need `duplicates`, why do we even need to cache them?

Reply:
i cached to solve that todo . The function didnt handle duplicates in descriptor .